tracing-core/src/dispatcher.rs

```diff
@@ -800,4 +807,38 @@ mod test {
 
         with_default(&Dispatch::new(TestSubscriber), || mk_span())
     }
+
+    #[test]
+    fn create_default_dispatcher() {
+        let default_dispatcher = Dispatch::default();
+        assert!(default_dispatcher.is::<NoSubscriber>());
+    }
+
+    #[cfg(feature = "std")]
+    #[test]
+    fn returns_default_dispatcher() {
+        struct TestSubscriber;
+        impl Subscriber for TestSubscriber {
+            fn enabled(&self, _: &Metadata<'_>) -> bool {
+                true
+            }
+
+            fn new_span(&self, _: &span::Attributes<'_>) -> span::Id {
+                span::Id::from_u64(0xAAAA)
+            }
+
+            fn record(&self, _: &span::Id, _: &span::Record<'_>) {}
+
+            fn record_follows_from(&self, _: &span::Id, _: &span::Id) {}
+
+            fn event(&self, _: &Event<'_>) {}
+
+            fn enter(&self, _: &span::Id) {}
+
+            fn exit(&self, _: &span::Id) {}
+        }
+        let _guard = set_default(&Dispatch::new(TestSubscriber));
+        let default_dispatcher = Dispatch::default();
+        assert!(default_dispatcher.is::<TestSubscriber>());
```

Comment:
I'd just add something to this test, like
```rust
// existing code...
drop(guard);
let default_dispatcher = Dispatch::default();
assert!(default_dispatcher.is::<NoSubscriber>());
```
Since the `guard` field is now used, the `_` should be removed.

This isn't a big deal, though — I'm happy to merge the PR without it.

Reply:
got it, makes a lot of sense to test that way, I've added that to the PR.